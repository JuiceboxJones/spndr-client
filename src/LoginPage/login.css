
.login_container{
  width: 30%;
  margin: 0 auto;
  margin-top: 10%;
  border-radius: 8px;
  background: #f5f5f5;
}

.login_container form{
  margin: 10px;
}

.login_container h2{
  background: #e0dfdf;;
  height: 32px;
  margin: 0;
  padding: 10px;
  border-radius: 8px 8px 0 0;
}

#login_form_button{
  margin: 15px;
  border: none;
  background: #9AB7BD;
  width: 200px;
  border-radius: 3px;
  height: 30px;
  font-weight: bold;
  cursor: pointer;
}

#login_form_button:hover{
  background: #68808B;
}