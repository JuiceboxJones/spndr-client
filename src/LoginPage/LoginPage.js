import React, { Component } from 'react';
import LoginForm from '../LoginForm/LoginForm';
import Header from '../Header/Header';
import './login.css';

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };

  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/welcome';
    history.push(destination);
  };

  render() {
    return (
      <div>
        <Header />
        <div className="login_container">
          <section className="LoginPage">
            <h2>Login</h2>
            <p>Demo Credentials</p>
            <p>Username: Test123</p>
            <p>Password: Password1!</p>
            <LoginForm onLoginSuccess={this.handleLoginSuccess} />
          </section>
        </div>
      </div>
    );
  }
}
