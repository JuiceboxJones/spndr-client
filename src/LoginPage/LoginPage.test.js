import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './LoginPage'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LoginPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<LoginPage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});