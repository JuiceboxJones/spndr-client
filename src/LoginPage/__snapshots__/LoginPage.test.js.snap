// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <header>
      <h1>
        Welcome to SPNDR
      </h1>
    </header>
    <p>
      Finance Management. Simplified.
    </p>
  </div>
  <div
    className="login_container"
  >
    <section
      className="LoginPage"
    >
      <h2>
        Login
      </h2>
      <form
        className="LoginForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="user_name"
        >
          <label
            htmlFor="LoginForm__user_name"
          >
            User name:
          </label>
          <input
            id="LoginForm__user_name"
            name="user_name"
            required={true}
          />
        </div>
        <div
          className="password"
        >
          <label
            htmlFor="LoginForm__password"
          >
            Password:
          </label>
          <input
            id="LoginForm__password"
            name="password"
            required={true}
            type="password"
          />
        </div>
        <button
          id="login_form_button"
          type="submit"
        >
          Login
        </button>
      </form>
    </section>
  </div>
</div>
`;
